<?xml version="1.0"?>
<launch>
  <arg name="algo" default="AStar" />
  <arg name="c1" default="true" />
  <arg name="c2" default="true" />

  <arg name="s1" default="-1.0" />
  <arg name="s2" default="1.0" />
  <arg name="s3" default="1.0" />
  <arg name="s4" default="1.0" />
  <arg name="s5" default="1.0" />

  <node pkg="path_follower" type="path_follower_node" name="path_follower" output="screen" respawn="true">
    <param name="supervisor/use_path_lookout" value="false" />
    <param name="supervisor/use_waypoint_timeout" value="false" />
    <param name="supervisor/use_distance_to_path" value="false" />

    <param name="supervisor/path_lookout/obstacle_scale_distance" value="1" />
    <param name="supervisor/path_lookout/obstacle_scale_lifetime" value="5" />
    <param name="supervisor/path_lookout/path_width" value="0.5" />
    <param name="supervisor/distance_to_path/max_dist" value="2.0" />
    
    <param name="controller" value="kinematic_SLP" />
    
    <param name="waypoint_tolerance" value="2.0" />
    <param name="goal_tolerance" value="0.05" />
    <param name="max_distance_to_path" value="2.0" />
    <param name="min_velocity" value="0.01" />
    <param name="max_velocity" value="0.6" />
    <param name="collision_box/width" value="0.4" />
    <param name="collision_box/crit_length" value="0.6" />
    <param name="collision_box/min_length" value="1.0" />
    <param name="collision_box/max_length" value="1.5" />
    <param name="collision_box/velocity_factor" value="1.0" />
    <param name="collision_box/velocity_saturation" value="1.0" />
    <param name="obstacle_scale_distance" value="1.0" />
    <param name="obstacle_scale_lifetime" value="5.0" />
    <param name="path_width" value="0.5" />
 
    <param name="k1" value="5.0" />
    <param name="k2" value="5.0" />

    <param name="gamma" value="5.0" />
    <param name="theta_a" value="4*0.78539816339" />
    <param name="max_angular_velocity" value="0.5" />
    <param name="epsilon" value="0.5" />
    <param name="b" value="0.2" />

    <param name="look_ahead_dist" value="0.5" />
    <param name="k_o" value="0" />
    <param name="k_g" value="0.1" />
    <param name="k_w" value="0.1" />
    <param name="k_curv" value="0.00" />

    <!-- local planner -->
    <param name="algo" value="$(arg algo)"/>
    <!-- constraints -->
    <param name="c1" value="$(arg c1)"/>
    <param name="c2" value="$(arg c2)"/>
    <!-- scorers -->
    <param name="s1" value="$(arg s1)"/>
    <param name="s2" value="$(arg s2)"/>
    <param name="s3" value="$(arg s3)"/>
    <param name="s4" value="$(arg s4)"/>
    <param name="s5" value="$(arg s5)"/>

    <remap from="scan" to="scan/front" />
    <remap from="/obstacles" to="/obstacle_cloud" />
  </node>
  
</launch>
